Chapter 5 carlos commments (#132)
chhoumann/p7: report/sections/Architecture/architecture.tex

The platform allows lecturers to create a syllabus that includes problem sets, exercises and test cases for the exercises. Additionally, students can do the exercises that the lecturer has provided.
The platform is a web application designed using a client-server architecture. Using this architecture the client communicates with a server using HTTP to send \textit{request} messages. The server waits for client requests and processes the requests when they arrive and replies to the requests with a \textit{response} message.

> The platform has a frontend component containing a user interface and logic to communicate to the backend component. 

Reviewer: 4 of the first 7 lines start with `the platform`, seems a bit excessive

Author: Det skal komme i andre rettelser, ikke dem fra carlos

The backend component contains business logic to handle user requests. 

> It supports role-based authentication with roles, meaning that users of varying roles have access to different functionality. 

Reviewer: ```suggestion
It supports role-based authentication, meaning that users have access to different functionality, depending on their role. 
```
Writing roles that many times seems redundant

	\label{fig:Architecture}
\end{figure}

\section{Backend System}

> The backend system has several responsibilities including routing, data retrieval and updates and orchestration of execution of code tests. 

Reviewer: ```suggestion
The backend system has several responsibilities including routing, data retrieval and mutations, and orchestration of execution of code tests. 
```
We use mutations in other places + have described it earlier in the report as being similar to creating/updating/etc. data.

The backend is the link between all the components in the system. It operates as a traditional server in a client-server architecture.

\section{Frontend} \label{sec:architecture-frontend}
The frontend is the user facing component of our web application. The frontend is designed such that its user interface and functionality is dependent on the role of the user.
For example if a student has signed in, the student is able to see and complete exercises. In turn if a lecturer signs in, they will have access to more functionality such as creating a syllabus or exercise.

> Regardless of the roles of the user the frontend has the ability to send data to the backend for storage, and show the output of executed code.

Reviewer: ```suggestion
Regardless of the user's role, the frontend has the ability to send data to the backend for storage, and show the output of executed code.
```

Users only have a single role
